index.d.ts

```diff
@@ -0,0 +1,182 @@
+// Type definitions for redux-little-router 14.0.0
+// Project: https://github.com/FormidableLabs/redux-little-router
+// Definitions by: priecint <https://github.com/priecint>
+
+import * as React from "react";
+import {Reducer, Middleware, StoreEnhancer} from "redux";
+
+export interface Routes {
+  [index: string]: {
+    [index: string]: RouteDefinition;
+  };
+}
+
+export interface RouteDefinition {
+  [index: string]: string | RouteDefinition;
+}
+
+export type LocationOptions = {
+  persistQuery?: boolean;
+};
+
+export type Query = { [key: string]: string };
+export type Params = { [key: string]: string };
+
+export type Location = {
+  basename?: string;
+  options?: LocationOptions;
+  params?: Params;
+  previous?: Location;
+  query?: Query;
+  result?: {};
```

Comment:
Would be good to have this as:
```
export type Location<TResult = {}> = {
  pathname?: string;
  options?: LocationOptions;
  params?: Params;
  previous?: Location;
  query?: Query;
  result?: TResult;
};
```

That way we can pass out interfaces to strongly type `result` when using it